cocos/2d/assembler/label/bmfontUtils.ts

```diff
@@ -39,34 +39,35 @@ import { view } from '../../../ui/view';
 const _defaultLetterAtlas = new LetterAtlas(64, 64);
 const _defaultFontAtlas = new FontAtlas(null);
 
-let _comp: Label | null = null;
-let _uiTrans: UITransform | null = null;
-
 let _fntConfig: IConfig | null = null;
 let _spriteFrame: SpriteFrame|null = null;
 let QUAD_INDICES;
 
 export const bmfontUtils = {
 
-    updateProcessingData (style: TextStyle, layout: TextLayout,
-        outputLayoutData: TextOutputLayoutData, outputRenderData: TextOutputRenderData,
-        comp: Label, trans: UITransform): void {
-        style.fontSize = comp.fontSize;
-        style.actualFontSize = comp.fontSize;
-        style.originFontSize = _fntConfig ? _fntConfig.fontSize : comp.fontSize;
-        layout.horizontalAlign = comp.horizontalAlign;
-        layout.verticalAlign = comp.verticalAlign;
-        layout.spacingX = comp.spacingX;
+    updateLayoutProcessingData (
+        style: TextStyle,
+        layout: TextLayout,
+        outputLayoutData: TextOutputLayoutData,
+        comp: Label,
+        trans: UITransform,
+    ): void {
+        style.fontSize = comp.fontSize; //both
+        style.actualFontSize = comp.fontSize; //both
+        style.originFontSize = _fntConfig ? _fntConfig.fontSize : comp.fontSize; //both
+        layout.horizontalAlign = comp.horizontalAlign; //both
```

Comment:
In design, style should only affect rendering, but because of the use of canvas, some rendering-related information is needed in the layout, and there is actually information that affects both rendering and layout, so it can't be completely separated.